### ISSUE TITLE
Support {fmt} v7.1.1

### ISSUE THREAD

**Issue by bl-ue** [odygrd/quill#63]
Recently, [{fmt} version 7.1.0](https://github.com/fmtlib/fmt/releases/tag/7.1.0) was released.

When using `quill` with vcpkg after microsoft/vcpkg#14217 and microsoft/vcpkg#14239, I am now experiencing the following issues when building:
```
quill.lib(FileUtilities.cpp.obj) : error LNK2019: unresolved external symbol "struct fmt::v7::detail::dragonbox::decimal_fp<float> __cdecl fmt::v7::detail::dragonbox::to_decimal<float>(float)" (??$to_decimal@M@dragonbox@detail@v7@fmt@@YA?AU?$decimal_fp@M@0123@M@Z) referenced in function "class fmt::v7::detail::buffer_appender<wchar_t> __cdecl fmt::v7::detail::write<wchar_t,class fmt::v7::detail::buffer_appender<wchar_t>,float,0>(class fmt::v7::detail::buffer_appender<wchar_t>,float)" (??$write@_WV?$buffer_appender@_W@detail@v7@fmt@@M$0A@@detail@v7@fmt@@YA?AV?$buffer_appender@_W@012@V3012@M@Z)
quill.lib(FileUtilities.cpp.obj) : error LNK2019: unresolved external symbol "struct fmt::v7::detail::dragonbox::decimal_fp<double> __cdecl fmt::v7::detail::dragonbox::to_decimal<double>(double)" (??$to_decimal@N@dragonbox@detail@v7@fmt@@YA?AU?$decimal_fp@N@0123@N@Z) referenced in function "class fmt::v7::detail::buffer_appender<wchar_t> __cdecl fmt::v7::detail::write<wchar_t,class fmt::v7::detail::buffer_appender<wchar_t>,double,0>(class fmt::v7::detail::buffer_appender<wchar_t>,double)" (??$write@_WV?$buffer_appender@_W@detail@v7@fmt@@N$0A@@detail@v7@fmt@@YA?AV?$buffer_appender@_W@012@V3012@N@Z)
```
...which I believe to be related to the switch from Grisu3 to Dragonbox for the default floating-point formatting, as quoted from the [release page](https://github.com/fmtlib/fmt/releases/tag/7.1.0):

>Switched from [Grisu3](https://www.cs.tufts.edu/~nr/cs257/archive/florian-loitsch/printf.pdf) to [Dragonbox](https://github.com/jk-jeon/dragonbox) for the default floating-point formatting which gives the shortest decimal representation with round-trip guarantee and correct rounding (#1882, #1887, #1894). 

I've opened an issue on vcpkg (microsoft/vcpkg#14331) thinking the issue was vcpkg, but now I realize that a new version of {fmt} was released and I think that is the problem.

**Comment by odygrd** [comment]
i updated the internal fmt version to 7.1 and it compiled without any problem. 
For vcpkg it gets the latest version of the library from vcpkg. 
Maybe quill was already installed with a lib fmt prior to 7.1.0 and then you updated lib fmt without re compiling quill ?
Can you try to rebuild quill via vcpkg please ?

**Comment by bl-ue** [comment]
@odygrd I'm sorry to ask again right after you update, but just for completeness will you please upgrade the internal fmt again?  [v7.1.1](https://github.com/fmtlib/fmt/releases/tag/7.1.1) was released just an hour ago.

**Comment by bl-ue** [comment]
Fixed with 6f2f6bc636c4c6e41dd4affc62cafa821d525c9c.
